File: A_The_Way_to_Home.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n, d;
    string s;
    cin >> n >> d >> s;

    int pos = 0 jumps = 0;
    while (pos < n - 1) {
        bool found = false;
        for (int i = min(n - 1, pos + d); i > pos; i--) {
            if (s[i] == '1') {
                pos = i;
                jumps++;
                found = true;
                break;
            }
        }
        if (!found) {
            cout << -1 << endl;
            return 0;
        }
    }

    cout << jumps << endl;
    return 0;
}

File: A_The_Way_to_Home.py
n, d = map(int, input().split())
s = input()

pos, jumps = 0, 0
while pos < n - 1:
    found = False
    for i in range(min(n - 1, pos + d), pos, -1):
        if s[i] == '1':
            pos = i
            jumps += 1
            found = True
            break
    if not found:
        print(-1)
        exit()

print(jumps)
